src/district/server/smart_contracts.cljs

```diff
@@ -192,19 +192,48 @@
                                     opts)
                              (fn [error event]
                                (callback error (web3-helpers/js->cljkk event))))))
-
-(defn contract-event-in-tx [contract event {:keys [:transaction-hash] :as tx-receipt}]
+; Original implementation
+(defn- get-event-from-tx-original
+  "Gets event decoded data from transaction receipt
+   Takes getter-fn which is a function that accepts predicate fn.
+     Example from Clojure standard library: some, filter"
+  [getter-fn contract event {:keys [:transaction-hash] :as tx-receipt}]
   (let [contract-instance (instance-from-arg contract)
         {:keys [:signature] :as event-interface} (web3-helpers/event-interface contract-instance event)]
     (promise-> (web3-eth/get-transaction-receipt @web3 transaction-hash)
                (fn [tx-receipt]
+                 (println "working w/ tx receipt " tx-receipt)
                  (let [{:keys [:logs :inputs]} (web3-helpers/js->cljkk tx-receipt)]
-                   (some (fn [{:keys [:topics :data] :as log}]
+                   (getter-fn (fn [{:keys [:topics :data] :as log}]
                            (when (= signature (first topics))
                              (let [return-values (web3-eth/decode-log @web3 (:inputs event-interface) data (drop 1 topics))]
+                               (println ">>>>> return-values are" return-values)
                                (web3-helpers/return-values->clj return-values event-interface))))
                     logs))))))
 
+(defn- get-events-from-tx
+  "Returns a promise that will resolve to the list of requested event data, decoded"
+  [contract event {:keys [:transaction-hash] :as tx-receipt} getter-fn]
+  (go
+    (let [contract-instance (instance-from-arg contract)
+          {:keys [:signature] :as event-interface} (web3-helpers/event-interface contract-instance event)
+          new-tx-receipt (<! (web3-eth/get-transaction-receipt @web3 transaction-hash))
+          {:keys [:logs :inputs]} (web3-helpers/js->cljkk new-tx-receipt)
+          sought-event? (fn [{:keys [:topics]}] (= signature (first topics)))
+          decode-event-data (fn [{:keys [:data :topics]}] (web3-eth/decode-log @web3 (:inputs event-interface) data (drop 1 topics)))
+          clojurize (fn [return-values] (web3-helpers/return-values->clj return-values event-interface))]
+      (->> logs
+           (filter sought-event? ,,,)
```

Comment:
Wonder what's the meaning of commas, is this some syntax sugar?

Reply:
I saw this convention in some place to mark where the threading macro puts the argument when it calls the method. Commas are treated like whitespace in Clojure.
With thread-first it would be
```clojure
(-> input
  (one-fn ,,, some-arg)
  (another-fn ,,, another-arg))
```